### ISSUE TITLE
Flags field

### ISSUE THREAD

**Issue by Sheep-y** [Sheep-y/ppdefmodifier#1]
As discussed on discord, I've made [some changes](https://github.com/tracktwo/ppdefmodifier/compare/tracktwo:99932f4...Sheep-y:8043adf) to support Modnix 3 Actions, to support plain text mods that mix mod info and ppdef commands.  Will cut a modnix beta release soon for easier testing.

The mod info and the commands can already be split.  That means we can have mod js at root, and a good old json in `PPDefModifier` folder.

So, the next step is, we need ppdefmodifier to know when a json should be processed traditionally, and when to wait for ActionMod.  Once that is done, mod authors can easily make one mod package that supports both new and old version of everything.

Adding a new "Flags" field to ModDefinition seems to be the best way.  A flag in the first def tells PPDef to ignore the file (when the stars align).  Minimal code change, while old ppdefmodifier will ignore the field and thus always load it.

The main drawback I see is that the json will be parsed twice, once by ppdefmodifier and once by Modnix.  Should be acceptable with the small number of tiny mods we have now.  When the mods go big we can implement faster detection, like read the first kb as text and run a regular expression.

Two alternatives I considered and discarded:
* Json naming scheme.  When a json is named as %mod_id%_action or %modid%action, and the stars align, it will be delegated.  Lightest code, fastest detection, but old mods need to be be manually deleted, and cannot be updated through overwrite.
* Smart auto detection.  About a thousand line of code, I guess.  Not a good direction for the mod.

**Comment by Sheep-y** [comment]
A little note.

After I wrote my thoughts down, I updated GUI and in the process changed the Action processing code to track which files are loaded as Actions.  This allows the GUI to track and manage those files.  Since the processing happens in the mod loader, it is available in-game and can be reterived through reflection, or may be a new api.

I almost went through and change course, before I realise that disabled mods will not have this information.

So, yeah, flag it is.
